leaf-ui/rappid-extensions/IntermediateValues.js

```diff
@@ -33,498 +33,327 @@ var IntermediateValuesTable = Backbone.View.extend({
-                                case '0000':
-                                    options = this.convertToOptions(['0000']);
-                                    break;
-                                case '0011':
-                                    options = this.convertToOptions(['0011']);
-                                    break;
-                                case '0100':
-                                    options = this.convertToOptions(['0100']);
-                                    break;
-                                case '1100':
-                                    options = this.convertToOptions(['1100']);
-                                    break;
-                                case '0010':
-                                    options = this.convertToOptions(['0010']);
-                                    break;
-                            }
-                        }
-                    } else if (func === "CR") {
-                        if (absVal < ti) {
-                            var funcX = intention.dynamicFunction.functionSegList[1].funcX;
-                            switch (funcX) {
-                                case '0000':
-                                    options = this.convertToOptions(['0000']);
-                                    break;
-                                case '0011':
-                                    options = this.convertToOptions(['0011']);
-                                    break;
-                                case '0100':
-                                    options = this.convertToOptions(['0100']);
-                                    break;
-                                case '1100':
-                                    options = this.convertToOptions(['1100']);
-                                    break;
-                                case '0010':
-                                    options = this.convertToOptions(['0010']);
-                                    break;
-                            }
-                        } else {
-                            var possibleValueList = ['0000', '0011', '0010', '1100', '0100', 'empty', 'no value'];
-                            options = this.convertToOptions(possibleValueList);
-                        }
-                    } else if (func === "SD") {
+    /**
+     * Iterates through all absolute time points
+     * And determines the appropriate range of value options
+     * Then passes those options to a SelectUserEvaluationView
+     * To be converted into a select dropdown and added to the table
+     */
+    loadSelect: function(){
 
-                        if (absVal < ti) {
-                            options = this.convertToOptions(['0011']);
-                        } else {
-                            options = this.convertToOptions(['1100']);
-                        }
+        // Iterate through all time points
+        for (let absTime of this.allAbsoluteTimePoints){
```

Comment:
So far this logic looks good, but I can't fully test until the other bug is fixed (see my other comment). This is so much simpler than what we had before, and dare I say more readable.